charts/consul/templates/server-config-configmap.yaml

```diff
@@ -72,6 +72,9 @@ data:
               "ca_file": "/consul/vault-ca/tls.crt",
               {{- end }}
               "intermediate_pki_path": "{{ .connectCA.intermediatePKIPath }}",
+              {{- if (and (.vaultNamespace) (not (contains "namespace" (default "" .connectCA.additionalConfig)))) }}
```

Comment:
json structure is like this https://github.com/hashicorp/consul-k8s/pull/2841/files#diff-73787600ab1d7b64c9a865fad8d8520d230644c998b3bf9b5670ffdb8900bb2eR713.
was having trouble fetching keys from nested array of json. hence had to keep it as string.
@zalimeni

Reply:
![Screenshot 2023-09-06 at 10 52 32 AM](https://github.com/hashicorp/consul-k8s/assets/134911583/6b02bdbe-2da8-4a8c-9ebe-fd6995df1dcb)
This is not the correct syntax. 
See - https://github.com/hashicorp/consul-k8s/blob/8f52b7c98e6f5b761c0ac69798c28ae152178de1/charts/consul/values.yaml#L234-L242
its inside json array.

Reply:
@absolutelightning ah, I'm sorry - that was unintuitive to me and I missed the `[]`s this whole time. (I also suspect looking at Consul code on the server side that it's not strictly necessary and this is an example of our JSON->HCL parsing complexity... 😞)

This looks overly complicated to detect in Helm template syntax even w/ the `first` template function as a sort-of option, so I feel a `"namespace"` `contains` check is a reasonable trade-off, given how unlikely it is someone would use that as a value (which would cause us to skip the `vaultNamespace` addition).

cc @thisisnotashwin in case you have any other thoughts, since we discussed this before.